### PR TITLE
Send out emails after database/firestore browser test failures.

### PR DESCRIPTION
**Pull request by yifanyang**
Since database/firestore SauceLabs tests are non-blocking, failures may not get noticed by developers. With this pr, in case of failure, an email notification containing the tested commit, summary of test results and link to the full build log, will be sent to the commit author.

A sample email looks like:

> Hey Yifan Yang,
> 
> Seems like you have pushed a commit:
> 
> commit 842f275c75c1d3e64cee28c24586a9d2d87eed0e
> Author: Yifan Yang <yifayan@gmail.com>
> Date:   Wed May 23 11:14:44 2018 -0700
> 
>     Send out emails after database/firestore browser test failures.
> 
> However, the database/firestore browser test has failed:
> 
> SUMMARY
>  0: Chrome 66.0.3359 (Windows 10.0.0): Executed 1357 of 1420 (skipped 63) SUCCESS (5 mins 30.833 secs / 4 mins 19.689 secs)
>  1: IE 11.0.0 (Windows 10.0.0): Executed 958 of 1002 (4 FAILED) (skipped 44) (3 mins 3.598 secs / 2 mins 55.564 secs)
>  2: Firefox 60.0.0 (Windows 10.0.0): Executed 1345 of 1420 (1 FAILED) (skipped 60) (5 mins 44.31 secs / 5 mins 30.632 secs)
>  3: Edge 17.17134.0 (Windows 10.0.0): Executed 958 of 1002 (skipped 44) SUCCESS (2 mins 44.473 secs / 2 mins 39.199 secs)
>  4: Safari 11.0.1 (Mac OS X 10.13.1): Executed 1345 of 1420 (5 FAILED) (skipped 60) (6 mins 4.427 secs / NaN secs)
>                                                                                  0  1  2  3  4 
>   Transaction Tests
>     server values: local timestamp should eventually (but not immediately) ma... ✔  ✔  ✔  ✔  ✖ 
>   QueryListener
>     raises error event                                                           ✔  ✔  ✔  ✔  ✖ 
>   Persistence:
>     (Memory) Visible mutations reflect uid switches                              ✔  ✖  ✔  ✔  ✔ 
>   Remote store:
>     (Memory) Handles user changes while offline (b/74749605).                    ✔  ✖  ✔  ✔  ✔ 
>   Writes:
>     (Memory) Writes are resent after network disconnect                          ✔  ✖  ✔  ✔  ✔ 
>   AsyncQueue
>     handles failures                                                             ✔  ✔  ✖  ✔  ✖ 
>     can schedule ops in the future                                               ✔  ✔  ✔  ✔  ✖ 
>   (Persistence=false) Queries
>     throws custom error when using docChanges as property                        ✔  ✖  ✔  ✔  ✔ 
>   1333 more test cases successful in all browsers
>  [32m24 05 2018 16:02:44.372:INFO [launcher.sauce]:  [39mShutting down Sauce Connect
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> 
> See full build logs at: https://travis-ci.org/firebase/firebase-js-sdk/builds/383256868.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,7 +35,7 @@ matrix:
 
 jobs:
   allow_failures:
-    - script: yarn test:saucelabs --database-firestore-only
+    - script: ./scripts/saucelabs/database_firestore_test.sh
```
Comment (mikelehen) [inline .travis.yml]: maybe name this "run_flaky_tests.sh" or something so we can add/remove tests without renaming the script?

File: scripts/saucelabs/report_test_failure.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# Sends out an email to author of the last commit with summary of the
+# browser test result.
+
+TRAVIS_BUILD_URL=https://travis-ci.org/$TRAVIS_REPO_SLUG/builds/$TRAVIS_BUILD_ID
+COMMITTER_EMAIL="$(git log -1 $TRAVIS_COMMIT --pretty="%cE")"
+AUTHOR_NAME="$(git log -1 $TRAVIS_COMMIT --pretty="%aN")"
+
+SUBJECT="[firebase-js-sdk] Browser Test Failed"
+
+echo "Hey $AUTHOR_NAME," > email.body
```
Comment (mikelehen) [inline scripts/saucelabs/report_test_failure.sh]: I'm not much of a bash expert, but I think you could write this as something like:

```
cat > email.body <<EOL
Hey $AUTHOR_NAME,

Seems like you have pushed a commit:

$(git log -1 $TRAVIS_COMMIT)
... 
EOL
```

File: scripts/saucelabs/report_test_failure.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+echo >> email.body
+echo "$(git log -1 $TRAVIS_COMMIT)" >> email.body
+echo >> email.body
+echo "However, the database/firestore browser test has failed:" >> email.body
+echo >> email.body
+sed -n "/SUMMARY/,$ p" browser_test.log >> email.body
+echo >> email.body
+echo "See full build logs at: $TRAVIS_BUILD_URL." >> email.body
+
+mail -s "$SUBJECT" -c yifany@google.com -- $COMMITTER_EMAIL < email.body
```
[the remark this answers is not in the document]
Comment (mikelehen) [reply scripts/saucelabs/report_test_failure.sh]: Hrm...  We have a couple aliases but they're internal-only I think...  You could create a new one that allows external email.  Or we can start with this for now and see how it goes.

File: .travis.yml (hunk 1 above)
Comment (yifanyang) [reply .travis.yml]: Done. I agree. This script (and flakiness measurement in the future) may not be limited to only database and firestore tests.

File: scripts/saucelabs/report_test_failure.sh (hunk 2 above)
Comment (yifanyang) [reply scripts/saucelabs/report_test_failure.sh]: This looks awesome!

**Comment by yifanyang** [comment]
Closing this pull request. PR #919 does the same thing in a more structured way.

**Pull request closed without merging**